src/platform/Linux/OTAImageProcessorImpl.cpp

```diff
@@ -165,19 +167,44 @@ void OTAImageProcessorImpl::HandleProcessBlock(intptr_t context)
         return;
     }
 
-    // TODO: Process block header if any
+    ByteSpan block   = imageProcessor->mBlock;
+    CHIP_ERROR error = imageProcessor->ProcessHeader(block);
 
-    if (!imageProcessor->mOfs.write(reinterpret_cast<const char *>(imageProcessor->mBlock.data()),
-                                    static_cast<std::streamsize>(imageProcessor->mBlock.size())))
+    if (error == CHIP_NO_ERROR &&
+        !imageProcessor->mOfs.write(reinterpret_cast<const char *>(block.data()), static_cast<std::streamsize>(block.size())))
     {
-        imageProcessor->mDownloader->EndDownload(CHIP_ERROR_WRITE_FAILED);
+        error = CHIP_ERROR_WRITE_FAILED;
+    }
+
+    if (error != CHIP_NO_ERROR)
+    {
+        imageProcessor->mDownloader->EndDownload(error);
         return;
     }
 
-    imageProcessor->mParams.downloadedBytes += imageProcessor->mBlock.size();
+    imageProcessor->mParams.downloadedBytes += block.size();
     imageProcessor->mDownloader->FetchNextData();
 }
 
+CHIP_ERROR OTAImageProcessorImpl::ProcessHeader(ByteSpan & block)
+{
+    if (mHeaderParser.IsInitialized())
+    {
+        OTAImageHeader header;
+        CHIP_ERROR error = mHeaderParser.AccumulateAndDecode(block, header);
+
+        // Needs more data to decode the header
+        ReturnErrorCodeIf(error == CHIP_ERROR_BUFFER_TOO_SMALL, CHIP_NO_ERROR);
+        ReturnErrorOnFailure(error);
+
+        mSoftwareVersion       = header.mSoftwareVersion;
+        mParams.totalFileBytes = header.mPayloadSize;
+        mHeaderParser.Clear();
+    }
```

Comment:
Suggest adding a comment that trusting/using the header's version is just an example to simplify host-based examples and is not required.

This is to avoid continued cut and paste of Linux behavior in other platforms.

It's important to note that the OTA header is *not* signed and there would only be authentication if its content was included in a further platform/product-specific image signature in the opaque payload of the image.

Reply:
Added a comment explaining the purpose of the code.